src/redux/books/booksSlice.js

```diff
@@ -1,42 +1,54 @@
 // bookSlice.js
-import { createSlice } from '@reduxjs/toolkit';
-
-const initialState = {
-  books: [
-    {
-      id: '1',
-      title: 'Game of thrones',
-      author: 'Nicholas',
-      category: 'Adventure',
-    },
-    {
-      id: '2',
-      title: 'God Father',
-      author: 'Kwamay',
-      category: 'Action',
-    },
-    {
-      id: '3',
-      title: 'Water Reunion',
-      author: 'Madaline Kiss',
-      category: 'Fantasy',
-    },
-  ],
-};
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const apiKey = 'uvoKogb5K4j0ehIhDE2a';
+const baseUrl = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/';
+
+const ADD_BOOK = 'books/booksSlice/ADD_BOOK';
+const REMOVE_BOOK = 'books/booksSlice/REMOVE_BOOK';
+const GET_BOOK = 'books/booksSlice/GET_BOOK';
+
+const initialState = [];
+
+export const addBook = createAsyncThunk(ADD_BOOK, async (book) => {
+  await axios.post(`${baseUrl}${apiKey}/books`, book);
+  return book;
+});
+
+export const removeBook = createAsyncThunk(REMOVE_BOOK, async (id) => {
+  const itemId = id;
+  await axios.delete(`${baseUrl}${apiKey}/books/${itemId}`, itemId);
+  return id;
+});
```

Comment:
- [ ] @Kwamay Kindly consider checking the implementation of the delete functionality, am unable to delete a book using the delete button.